service/qos/service_level_controller.cc

```diff
@@ -298,6 +298,22 @@ void service_level_controller::update_from_distributed_data(std::chrono::duratio
     }
 }
 
+future<> service_level_controller::stop_update_from_distributed_data() {
+    if (this_shard_id() != global_controller) {
+        return make_ready_future();
+    }
+    // abort the loop of the distributed data checking if it is running
+    if (!_global_controller_db->distributed_data_update.available()) {
+        if (!_global_controller_db->dist_data_update_aborter.abort_requested()) {
+            _global_controller_db->dist_data_update_aborter.request_abort();
```

Comment:
you don't need to wrap `request_abort` in a check - it's an idempotent operation, so you can just call it unconditionally

Reply:
Will change.